### PR TITLE
Features/touchups

### PR DESCRIPTION
**Pull request by seanmckay94**

### CODE REVIEW

File: routes/auth/auth.js [hunk 1]
```diff
@@ -118,4 +118,18 @@ router.post("/login", async (req, res) => {
   }
 });
 
+router.delete("/logout", (req, res) => {
+  if (req.session) {
```
Comment (strahinjaajvaz) [inline routes/auth/auth.js]: thats wrong. you're not using sessions anymore...
This is WRONG!

you should be clearing the cookie... I went over this with you

File: public/js/logout.js [hunk 2]
```diff
@@ -1,15 +1,14 @@
...
+  if (response.ok) {
+    document.location.replace("/");
+  } else {
+    alert("Failed to log out.");
+  }
 };
 
-document.querySelector('#logout').addEventListener('click', logout);
\ No newline at end of file
+document.getElementById("#logout").addEventListener("click", logout);
```
Comment (strahinjaajvaz) [inline public/js/logout.js]: you have jQuery, why are you doing this!

File: routes/html/index.js [hunk 3]
```diff
@@ -1,20 +1,27 @@
 const Users = require("../../models/Users");
 const Games = require("../../models/Games");
 const sequelize = require("../../config/connection");
+const { withAuth } = require("../../utils/auth");
 
 const router = require("express").Router();
 
-router.get("/", (req, res) => {
+router.get("/", withAuth, async (req, res) => {
   const loggedIn = req.user;
```
Comment (strahinjaajvaz) [inline routes/html/index.js]: this is all wrong.

`req.user` isn't logged in. Its a User object. why are you calling it logged in?

File: routes/html/index.js [hunk 4]
```diff
@@ -1,20 +1,27 @@
...
-  if (loggedIn) {
+  if (!!loggedIn) {
     // render the dashboard
-    res.render("dashboard", { loggedIn });
+    const fetchName = req.user.firstName;
```
Comment (strahinjaajvaz) [inline routes/html/index.js]: if the variable is logged in (implies its boolean), then how are you able to get some values from it?

File: routes/html/index.js [hunk 5]
```diff
@@ -1,20 +1,27 @@
...
+    const fetchUsername = req.user.username;
+    res.render("dashboard", { fetchName, fetchUsername, loggedIn });
```
Comment (strahinjaajvaz) [inline routes/html/index.js]: why are you passing a user object in as loggedIn? whats the purpose of it!

File: routes/html/index.js [hunk 6]
```diff
@@ -1,20 +1,27 @@
...
     res.render("landingpage");
   }
 });
 
+router.get("/landing", (req, res) => {
+  res.render("landingpage");
+});
+
 router.get("/login", (req, res) => {
   const loggedIn = req.user;
```
Comment (strahinjaajvaz) [inline routes/html/index.js]: same as above. if you copy code blindly, you copy along the mistakes...

File: routes/html/index.js [hunk 7]
```diff
@@ -86,7 +93,13 @@ router.get("/games", async (req, res) => {
 });
 
 router.get("/about-us", (req, res) => {
-  // returns an about us page with some information on there
+  const loggedIn = req.user;
```
Comment (strahinjaajvaz) [inline routes/html/index.js]: and here

File: routes/html/index.js [hunk 8]
```diff
@@ -86,7 +93,13 @@ router.get("/games", async (req, res) => {
...
+  res.render("about-us", { loggedIn });
+});
+
+router.get("/logout", (req, res) => {
+  const loggedIn = req.user;
```
Comment (strahinjaajvaz) [inline routes/html/index.js]: and here

File: views/dashboard.handlebars [hunk 9]
```diff
@@ -1,4 +1,26 @@
...
 
+<div class="container">
+    <div class="hero ">
+        <div class="hero-body" >
+            <img src="/images/gamers_game.png" alt="banner with gaming consol controllers">
+        </div>
+    </div>
+    <div class="hero is-primary has-text-black is-medium">
+        <div class="hero-body has-text-centered">
+            <h1 class="title"> Hello {{fetchName}} !</h1>
```
Comment (strahinjaajvaz) [inline views/dashboard.handlebars]: fetch name is a misleading variable. it imples that youre going to fetch something. it should be name of full name

**Review by strahinjaajvaz — CHANGES_REQUESTED** [review]

File: routes/html/index.js (hunk 3 above)
Comment (tinku-bansal) [reply routes/html/index.js]: fixed in other PR

File: routes/html/index.js (hunk 5 above)
Comment (tinku-bansal) [reply routes/html/index.js]: fixed in other PR

File: routes/html/index.js (hunk 6 above)
Comment (tinku-bansal) [reply routes/html/index.js]: fixed in other PR

File: routes/auth/auth.js (hunk 1 above)
Comment (KorolyovaLara) [reply routes/auth/auth.js]: Fixed

File: public/js/logout.js (hunk 2 above)
Comment (KorolyovaLara) [reply public/js/logout.js]: $("#logoutBtn").on("click", logout);

File: views/dashboard.handlebars (hunk 9 above)
Comment (KorolyovaLara) [reply views/dashboard.handlebars]: Renamed 
userDBname, userDBusername

**Comment by KorolyovaLara** [comment]
Logout is fixed and few other small issues with GitHub logo and Intro not getting saved due to  misspell. 
Removed favicon. 

Apologies for large change. 

Next on the list to fix : 

- [x] navbar not working in small version
- [ ] user-page not inheriting styling

File: public/js/logout.js [hunk 10]
```diff
@@ -1,15 +1,17 @@
...
+  const response = await fetch("/auth/logout", {
+    method: "DELETE",
+    headers: {
+      Accept: "application/json",
+      "Content-Type": "application/json",
+    },
+  });
+  if (response.ok) {
+    window.location.href = "/";
+    return;
```
Comment (strahinjaajvaz) [inline public/js/logout.js]: No need for return.

File: routes/api/gameApi.js [hunk 11]
```diff
@@ -28,7 +28,7 @@ router.get("/", withAuth, async (req, res) => {
       res.json(games);
     } catch (e) {
       console.log(e);
```
Comment (strahinjaajvaz) [inline routes/api/gameApi.js:30]: Remove

File: routes/api/gameApi.js [hunk 12]
```diff
@@ -28,7 +28,7 @@ router.get("/", withAuth, async (req, res) => {
...
-      return res.status(409).json({ message: `Unexpected Error!` });
+      return res.status(500).json({ message: "Unexpected Error!" });
     }
   } catch (e) {
     console.log(e);
```
Comment (strahinjaajvaz) [inline routes/api/gameApi.js:34]: Remove

File: routes/html/index.js [hunk 13]
```diff
@@ -1,20 +1,27 @@
...
   const loggedIn = !!req.user;
   if (loggedIn) {
     // render the dashboard
-    res.render("dashboard", { loggedIn });
+    const userDBname = req.user.firstName;
+    const userDBusername = req.user.username;
+    res.render("dashboard", { userDBname, userDBusername, loggedIn });
   } else {
     // render the landing page
     res.render("landingpage");
```
Comment (strahinjaajvaz) [inline routes/html/index.js:14]: Shouldn't this redirect to the landing page?
You have two routes rendering to the same screen

File: views/dashboard.handlebars [hunk 14]
```diff
@@ -1,4 +1,26 @@
...
 
+<div class="container">
+    <div class="hero ">
+        <div class="hero-body" >
+            <img src="/images/gamers_game.png" alt="banner with gaming consol controllers">
+        </div>
+    </div>
+    <div class="hero is-primary has-text-black is-medium">
+        <div class="hero-body has-text-centered">
+            <h1 class="title"> Hello {{userDBname}} !</h1>
```
Comment (strahinjaajvaz) [inline views/dashboard.handlebars]: What's wrong with just username?

File: views/dashboard.handlebars [hunk 15]
```diff
@@ -1,4 +1,26 @@
...
+            <h2 > Long time no see... or was it just yesterday?!</h2>
+            <h2> We are playing so many games that we lost track of time here...oops.</h2>
+            <h2> But at least you are on track.</h2>
+            <h2> Check your profile and update your games.</h2>
+        </div>
+        <div class="button is-large is-danger">
+            <a href="/user/{{userDBusername}}">
```
Comment (strahinjaajvaz) [inline views/dashboard.handlebars]: Same as above.

Names should be simple

File: routes/html/index.js (hunk 13 above)
Comment (KorolyovaLara) [reply routes/html/index.js:14]: Dashboard is page where user can access their profile link to their page

Landing page is our logo/banner page

File: views/dashboard.handlebars (hunk 14 above)
Comment (KorolyovaLara) [reply views/dashboard.handlebars]: isn't bad if we are using same const names all over again?!
This way it it is kind "info from DB"

**Review by KorolyovaLara — COMMENTED** [review]
all fixed according with request changes

File: views/dashboard.handlebars [hunk 16]
```diff
@@ -1,4 +1,26 @@
 {{> nav}}
 
+<div class="container">
+    <div class="hero ">
+        <div class="hero-body" >
+            <img src="/images/gamers_game.png" alt="banner with gaming consol controllers">
```
Comment (seanmckay94) [inline views/dashboard.handlebars:6]: Small spelling error in alt tag ( consol )

File: views/logout.handlebars [hunk 17]
```diff
@@ -0,0 +1,24 @@
+{{> nav }}
+
+
+<div class="container">
+    <div class="hero-body level-item" >
+        <img src="/images/gamers_game.png" alt="banner with gaming consol controllers">
```
Comment (seanmckay94) [inline views/logout.handlebars:6]: Same minor spelling error ( consol )

File: views/logout.handlebars [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+        <div class="level-item">
+            <h1 class="title"> You want to Log Out? </h1>
+        </div>
+        <div class="button is-large is-danger" id="logoutBtn">
+          YES
+        </div>
+    </div>
+
+    <div class="hero-body level-item" >
+        <img src="/images/gamers_game.png" alt="banner with gaming consol controllers">
```
Comment (seanmckay94) [inline views/logout.handlebars:19]: Same minor spelling error ( consol )

**Pull request merged**